lib/ZwaveClient.ts

```diff
@@ -0,0 +1,3276 @@
+	ignoreDiscovery?: boolean
+	fan_mode_map?: { [key: string]: number }
+	mode_map?: { [key: string]: number }
+	id?: string
+}
+
+export type Z2MNode = {
+	id: number
+	manufacturerId?: number
+	productId?: number
+	productLabel?: string
+	productDescription?: string
+	productType?: number
+	manufacturer?: string
+	firmwareVersion?: string
+	protocolVersion?: ProtocolVersion
+	zwavePlusVersion?: number | undefined
+	zwavePlusNodeType?: ZWavePlusNodeType | undefined
+	zwavePlusRoleType?: ZWavePlusRoleType | undefined
+	nodeType?: NodeType
+	endpointsCount?: number
+	endpointIndizes?: number[]
+	isSecure?: boolean | 'unknown'
+	supportsBeaming?: boolean
+	supportsSecurity?: boolean
+	isListening?: boolean
+	isControllerNode?: boolean
+	isFrequentListening?: FLiRS
+	isRouting?: boolean
+	keepAwake?: boolean
+	deviceClass?: Z2MDeviceClass
+	neighbors?: number[]
+	loc?: string
+	name?: string
+	hassDevices?: { [key: string]: HassDevice }
+	deviceId?: string
+	hexId?: string
+	values?: { [key: string]: Z2MValueId }
+	groups?: Z2MNodeGroups[]
+	ready: boolean
+	available: boolean
+	failed: boolean
+	lastActive?: number
+	dbLink?: string
+	maxDataRate?: DataRate
+	interviewStage?: keyof typeof InterviewStage
+	status?: keyof typeof NodeStatus
+	inited: boolean
+	healProgress?: string | undefined
+}
+
+export type ZwaveConfig = {
+	port: string
+	networkKey: string
+	serverEnabled: boolean
+	serverPort: number
+	logEnabled: boolean
+	logLevel: LogManager.LogLevel
+	commandsTimeout: number
+	enableStatistics: boolean
+	disclaimerVersion: number
+	options: ZWaveOptions
+	healNetwork: boolean
+	healHour: number
+	logToFile: boolean
+	nodeFilter: string[]
+}
+
+export type Z2MDriverInfo = {
+	uptime?: number
+	lastUpdate?: number
+	status?: ZwaveClientStatus
+	cntStatus?: string
+	appVersion?: string
+	zwaveVersion?: string
+	serverVersion?: string
+	homeid?: number
+	name?: string
+	controllerId?: number
+	newConfigVersion?: string | undefined
+}
+
+export enum ZwaveClientStatus {
+	CONNECTED = 'connected',
+	DRIVER_READY = 'driver ready',
+	SCAN_DONE = 'scan done',
+	DRIVER_FAILED = 'driver failed',
+	CLOSED = 'closed',
+}
+
+export enum EventSource {
+	DRIVER = 'driver',
+	CONTROLLER = 'controller',
+	NODE = 'node',
+}
+
+export interface ICallApiResult {
+	message: any
+	args?: any
+	success?: boolean
+	result?: any
+}
+
+declare interface ZwaveClient {
+	on(event: 'nodeStatus', listener: (node: Z2MNode) => void): this
+	on(
+		event: 'event',
+		listener: (source: EventSource, eventName: string, ...args: any) => void
+	): this
+	on(event: 'scanComplete', listener: () => void): this
+	on(
+		event: 'notification',
+		listener: (node: Z2MNode, valueId: Z2MValueId, data: any) => void
+	): this
+	on(event: 'nodeRemoved', listener: (node: Z2MNode) => void): this
+	on(
+		event: 'valueChanged',
+		listener: (valueId: Z2MValueId, node: Z2MNode) => void
+	): this
+	on(
+		event: 'valueWritten',
+		listener: (valueId: Z2MValueId, value: any) => void
+	): this
+}
```

Comment:
You might want to define the other EventEmitter functions too:
https://github.com/zwave-js/node-zwave-js/blob/4b0ce9b613e63a683b25536456cb04ec7e0bb646/packages/zwave-js/src/lib/driver/Driver.ts#L324-L355

Reply:
Humm wait but If I extend EventEmitter in my class I already have those methods. At least I see them now